theseus_gui/src/pages/Settings.vue

```diff
@@ -1,7 +1,471 @@
+      <div class="settings-group">
+        <h3>Console</h3>
+        <div class="toggle-setting">
+          Maximum Concurrent Downloads
+          <input
+            v-model="settings.max_concurrent_downloads"
+            type="number"
+            name="concurrent-downloads"
+            class="concurrent-downloads"
+          />
+        </div>
+      </div>
+    </Card>
+    <Card class="settings-card">
+      <h2 class="settings-title">Commands</h2>
+      <div class="settings-group">
+        <div class="toggle-setting">
+          Pre Launch
+          <input ref="javaArgs" v-model="settings.hooks.pre_launch" type="text" class="input" />
+        </div>
+        <div class="toggle-setting">
+          Wrapper
+          <input ref="javaArgs" v-model="settings.hooks.wrapper" type="text" class="input" />
+        </div>
+        <div class="toggle-setting">
+          Post Launch
+          <input ref="javaArgs" v-model="settings.hooks.post_exit" type="text" class="input" />
+        </div>
+      </div>
+    </Card>
+    <Button color="primary" class="save-btn" @click="saveSettings"><SaveIcon />Save changes</Button>
   </div>
 </template>
+
+<style lang="scss">
+.active {
+  background: var(--color-brand-highlight) !important;
+}
+
+.concurrent-downloads {
+  min-height: 40px;
+  max-width: 4.2rem !important;
+  text-align: center;
+}
+
+.auto-detect-modal {
+  .modal-body {
+    width: 45rem !important;
+    .content {
+      padding: 1rem;
+
+      button {
+        width: 100%;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+
+        span {
+          display: inherit;
+          align-items: center;
+          justify-content: center;
+        }
+      }
+    }
+  }
+}
+
+.save-btn {
+  margin: 1rem;
+}
+
+.slider-input {
+  width: 5rem !important;
+  flex-basis: 5rem !important;
+}
+
+.installation-input {
+  width: 100% !important;
+  flex-grow: 1;
+}
+
+.theming,
+.settings-card {
+  margin: 1rem;
+  display: flex;
+  flex-direction: column;
+}
+
+.theme-dropdown {
+  text-transform: capitalize;
+}
+
+.settings-card {
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+}
+
+.settings-title {
+  color: var(--color-contrast);
+}
+
+.settings-group {
+  display: flex;
+  flex-direction: column;
+  gap: 0.5rem;
+}
+
+.installation-input {
+  width: 100%;
+}
+
+.installation-buttons {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  align-items: center;
+  gap: 0.5rem;
+  margin: 0;
+}
+
+.sliders {
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row;
+  gap: 1rem;
+  width: 100%;
+
+  .slider {
+    flex-grow: 1;
+  }
+}
+
+.toggle-setting {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  gap: 0.5rem;
+}
+
+.card-divider {
+  background-color: var(--color-button-bg);
+  border: none;
+  color: var(--color-button-bg);
+  height: 1px;
+  margin: var(--gap-sm) 0;
+}
```

Comment:
if this is reused, it should be moved to omorphia or global styles instead of duplicating it